google/cloud/bigtable/data/read_rows_query.py

```diff
@@ -123,7 +174,35 @@ def __bool__(self) -> bool:
         Empty RowRanges (representing a full table scan) are falsy, because
         they can be substituted with None. Non-empty RowRanges are truthy.
         """
-        return self.start is not None or self.end is not None
+        return self._start is not None or self._end is not None
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, RowRange):
+            return NotImplemented
+        return self._start == other._start and self._end == other._end
+
+    def __str__(self) -> str:
+        """
+        Represent range as a string, e.g. "[b'a', b'z)"
+        Unbounded start or end keys are represented as "-inf" or "+inf"
+        """
+        left = "[" if self.start_is_inclusive else "("
+        right = "]" if self.end_is_inclusive else ")"
+        start = repr(self.start_key) if self.start_key is not None else "-inf"
+        end = repr(self.end_key) if self.end_key is not None else "+inf"
+        return f"{left}{start}, {end}{right}"
```

Comment:
Can we normalize the open bounds? ie we shouldn't have both (-inf and [-inf

Reply:
Perhaps the ctor can ensure that {start,end}_is_inclusive is deterministically set when the value is open ended

Reply:
`_is_inclusive` will be None when the range is None, so it will currently always be "(" and ")"

If we want to say that unbounded points always count as either inclusive or exclusive, that could make things simpler. But I wasn't sure if that could have downstream consequences

Reply:
Downstream it sholdnt matter because it will end up in the unset bound oneof case

Reply:
Ok, I made empty ranges count as inclusive